===================================================================
RCS file: /cvs/gnome/gnome-vfs/libgnomevfs/gnome-vfs-hal-mounts.c,v
retrieving revision 1.29.2.1
retrieving revision 1.29.2.2
diff -u -r1.29.2.1 -r1.29.2.2
--- gnome-vfs-hal-mounts.c	2006/04/25 12:01:14	1.29.2.1
+++ libgnomevfs/gnome-vfs-hal-mounts.c	2006/05/11 14:43:10	1.29.2.2
@@ -793,6 +793,8 @@
 	GnomeVFSVolumeMonitor *volume_monitor;
 	GnomeVFSHalUserData *hal_userdata;
 	char *name;
+	DBusError error;
+	gboolean media_check_enabled;
 
 	g_return_if_fail (hal_drive != NULL);
 
@@ -828,6 +830,18 @@
 		goto out;
 	}
 	
+	dbus_error_init (&error);
+ 	media_check_enabled = libhal_device_get_property_bool (volume_monitor_daemon->hal_ctx,
+ 							       libhal_drive_get_udi (hal_drive),
+ 							       "storage.media_check_enabled",
+ 							       &error);
+ 	if (dbus_error_is_set (&error)) {
+ 		g_warning ("Error retrieving storage.media_check_enabled on '%s': Error: '%s' Message: '%s'",
+ 			   libhal_drive_get_udi (hal_drive), error.name, error.message);
+ 		dbus_error_free (&error);
+ 		media_check_enabled = FALSE;
+ 	}
+	
 	drive = g_object_new (GNOME_VFS_TYPE_DRIVE, NULL);
 	drive->priv->activation_uri = g_strdup ("");
 	drive->priv->is_connected = 1;
@@ -840,7 +854,7 @@
 	name = g_utf8_casefold (drive->priv->display_name, -1);
 	drive->priv->display_name_key = g_utf8_collate_key (name, -1);
 	g_free (name);
-	drive->priv->is_user_visible = TRUE;
+	drive->priv->is_user_visible = !media_check_enabled; /* See http://bugzilla.gnome.org/show_bug.cgi?id=321320 */
 	drive->priv->volumes = NULL;
 	drive->priv->hal_udi = g_strdup (libhal_drive_get_udi (hal_drive));
         drive->priv->must_eject_at_unmount = libhal_drive_requires_eject (hal_drive);
